store/ENSSlice/index.ts

```diff
@@ -0,0 +1,58 @@
+import { ethers } from 'ethers';
+import { StateSlice } from '~/store/types';
+import { IENSSlice } from './types';
+import { StoreState } from '..';
+
+const TTL = 86400 * 1000; // 1 day in milliseconds
+
+const storeENSName = (account: string, name: string): void => {
+    const ensName = {
+        name,
+        expiry: Date.now() + TTL,
+    };
+    localStorage.setItem(ethers.utils.getAddress(account), JSON.stringify(ensName));
+};
+
+const getCachedENS = (account: string): string | null => {
+    const nameStr = localStorage.getItem(ethers.utils.getAddress(account));
+    if (!nameStr) {
+        return null;
+    }
+    const ensName = JSON.parse(nameStr);
+    const now = Date.now();
+    if (now > ensName.expiry) {
+        localStorage.removeItem(ethers.utils.getAddress(account));
+        return null;
```

Comment:
could potentially simplify here by making this function set the ens name in the cache if its expired or does not exist and not needing `storeENSName`

Reply:
not entirely sure what you mean by this? restoring if its expired?

Reply:
yeah pretty much just putting the logic of `storeENSName` here and returning the result. 

so the behaviour of `getCachedENS` is actually more like `get the cached one or populate the cache if required and return the result`

Reply:
but then say no cached items exist id have to
getCached -> provider.lookupAddress -> getCached or always provider.lookupAddress -> getCached